relay-server/src/services/project_cache.rs

```diff
@@ -899,6 +865,42 @@ impl ProjectCacheBroker {
         }
     }
 
+    /// Returns `true` if the project state valid for the [`QueueKey`].
+    ///
+    /// Which includes the own key and the samplig key for the project.
+    fn is_state_valid(&mut self, key: &QueueKey) -> bool {
+        let QueueKey {
+            own_key,
+            sampling_key,
+        } = key;
+
+        let is_own_state_valid = self.projects.get_mut(own_key).map_or(false, |project| {
+            // Returns `Some` if the project is cache otherwise None and also triggers refresh
+            // in background.
+            project
+                .get_cached_state(self.services.project_cache.clone(), false)
+                // Makes sure that the state also is valid.
+                .map_or(false, |state| !state.invalid())
+        });
+
+        let is_sampling_state_valid = if own_key != sampling_key {
+            self.projects
+                .get_mut(sampling_key)
+                .map_or(false, |project| {
+                    // Returns `Some` if the project is cache otherwise None and also triggers refresh
+                    // in background.
```

Comment:
```suggestion
                    // Returns `Some` if the project is cached otherwise None and also triggers a refresh
                    // in the background.
```